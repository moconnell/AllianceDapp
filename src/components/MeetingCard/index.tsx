import Meeting from "../../types/meeting";
import { formatDateTime, formatTime } from "../../utils/formatDate";
import {
  DateText,
  CardContainer,
  MeetingDescription,
  Attendee,
} from "./styles";

interface MeetingCardProps {
  meeting: Meeting;
}

const MeetingCard = ({
  meeting: { date, endDate, attendee, description },
}: MeetingCardProps) => (
  <CardContainer>
    <DateText>
      {formatDateTime(date)}-{formatTime(endDate)}
    </DateText>
    <Attendee>{attendee}</Attendee>
    <MeetingDescription>{description}</MeetingDescription>
  </CardContainer>
);

export default MeetingCard;
