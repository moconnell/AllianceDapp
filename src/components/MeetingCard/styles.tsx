// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'styl... Remove this comment to see the full error message
import styled from "styled-components";

export const CardContainer = styled.div`
  max-width: 200px;
  width: 196px;
  height: 170px;
  max-height: 200px;
  display: flex;
  justify-content: left;
  flex-wrap: wrap;
  background: #ff3864;
  border: 1px solid #ff3864;
  box-sizing: border-box;
  border-radius: 6px;
  padding: 6px 8px;
  margin-bottom: 6px;
`;
export const Attendee = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 19px;
  color: #fff;
`;

export const MeetingDescription = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #fff;
`;

export const DateText = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 28px;
  color: #fff;
`;
