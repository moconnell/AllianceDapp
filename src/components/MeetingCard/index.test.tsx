import { render } from "@testing-library/react";
import MeetingCard from "./index";

describe("<MeetingCard />", () => {
  it("should render correctly", async () => {
    const meeting = {
      date: new Date(Date.parse("2022-02-02 20:00")),
      endDate: new Date(Date.parse("2022-02-02 20:30")),
      attendee: "Dave Spart",
      description: "Random meeting",
    };
    const { asFragment } = render(<MeetingCard meeting={meeting} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
