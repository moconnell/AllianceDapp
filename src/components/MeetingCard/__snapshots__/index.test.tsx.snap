// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetingCard /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ lhfvPN"
  >
    <span
      class="sc-hKMtZM AizOZ"
    >
      Wednesday, February 2, 2022 at 8:00 PM-8:30 PM
    </span>
    <span
      class="sc-gsnTZi bBjkky"
    >
      Dave Spart
    </span>
    <span
      class="sc-dkzDqf ezVsXa"
    >
      Random meeting
    </span>
  </div>
</DocumentFragment>
`;
